### PR TITLE
fix(catalogue): add missing resource types and otherwise use raw type name

### PR DESCRIPTION
**Pull request by mswertz**
Closes https://github.com/molgenis/molgenis-emx2/issues/4342

Using data from https://data-catalogue-acc.molgeniscloud.org/catalogue/ssr-catalogue

Before:
image

After:
image

### CODE REVIEW

File: apps/nuxt3-ssr/utils/CollectionUtils.ts [hunk 1]
```diff
@@ -44,10 +46,10 @@ export function getResourceMetadataForType(
     Object.values(typeMetadata).filter(
       (value: IResourceTypeMetadata) => value.type === type
     )?.[0] || {
-      type: "Resource",
-      plural: "Resources",
+      type: type,
+      plural: type,
       image: "image-link",
-      path: "resources",
+      path: type,
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/nuxt3-ssr/utils/CollectionUtils.ts:52]: I notice  gap which is the descriptions are still missing

**Comment by mswertz** [comment]
> https://preview-emx2-pr-4348.dev.molgenis.org/catalogue-demo/ssr-catalogue/all
> 
> shows 'clinical trail' instead of expected 'clinical trails' , the paths ( routes of the other types are , all lowercase no whitespace ( for example ssr-catalogue/all/datasources ) but clinical trail has a path of 'sr-catalogue/all/Clinical%20trial'

Indeed. If all is well this is now fixed.
Also enable special rule for resourceType == 'resources' which we will need to show all results.

File: apps/nuxt3-ssr/components/header/Catalogue.vue [hunk 2]
```diff
@@ -29,7 +29,7 @@ if (route.params.resourceType) {
 if (props.resourceTypes.length > 0) {
   props.resourceTypes.forEach((resourceType) => {
     const resourceTypeMetadata = getResourceMetadataForType(
-      resourceType.type.name
+      resourceType.type?.name
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/nuxt3-ssr/components/header/Catalogue.vue:32]: Yes, we should allow for this to happen.

**Review by hslh — APPROVED** [review]
Looks good to me on the preview server. Perhaps out of the scope of this PR, but the label below the resource logo and count is still default 'Cohorts & Biobanks'. Perhaps change this to use the raw type name instead so it at least matches the name?

**Comment by mswertz** [comment]
> ut the label below the resource logo and count is still default 'Cohorts & Biobanks'. Perhaps change this to use the raw type name instead so it at least matches the name?

indeed. done.

**Pull request merged**